corehq/apps/app_manager/fixtures/mobile_ucr.py

```diff
@@ -133,12 +133,22 @@ class ReportDataCache(object):
     def __init__(self):
         self.data_cache = {}
         self.total_row_cache = {}
+        self.reports = {}
 
     def get_data(self, key, data_source):
         if key not in self.data_cache:
             self.data_cache[key] = data_source.get_data()
         return self.data_cache[key]
 
+    def fetch_reports(self, domain, report_configs):
+        """Bulk fetch all reports for syncing"""
+        report_ids = [config.report_id for config in report_configs]
+        self.reports = _get_reports_and_data_source(report_ids, domain)
```

Comment:
There's a weird ordering dependency here around setting `self.reports` that would be nice to avoid if doable.  Would it be possible to make `report_configs` and `domain` required parameters to `__init__`, and then make `self.reports` a `cached_property`?  I think that'd mean moving the initialization from `BaseReportFixtureProvider.__init__` to `__call__`.

Reply:
I added a few commits to update this:

1. pass all report configs to cache from the start
2. still call 'load' in the fixture provider to allow for error handling
3. load reports just in time if needed (purely a safeguard)
4. allow loading a subset of the reports to avoid loading unnecessary reports in the V2 only use case.

Reply:
thanks, looks great!